[bug] yges_c.a の変更が発生するときビルド通らない

```
gcc -o trash/test_never.exe -L/usr/local/lib -L../.. trash/test_never.o -l:yges_c.a -l:test/test.a
/usr/bin/ld: ../../test/test.a: error adding symbols: malformed archive
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:14: trash/test_never.exe] Error 1
```

--- a/test/lib/Makefile
+++ b/test/lib/Makefile
@@ -11,7 +11,7 @@ OBJS   = trash/dummy2.o
 all: $(TARGET)
 
 clean:
-	rm -f $(TARGET) $(OBJS)
+	rm -f $(TARGET) $(INTERMED) $(OBJS)
 
 $(TARGET): $(INTERMED) $(LINKAGES)
 	rm -f $(TARGET)

--- a/test/lib/Makefile
+++ b/test/lib/Makefile
@@ -4,6 +4,8 @@ LDFLAGS       = -L/usr/local/lib
 AR            = ar
 
 TARGET = ../test.a
+INTERMED = test_intermediate.a
+LINKAGES = ../../yges_c.a
 OBJS   = trash/dummy2.o 
 
 all: $(TARGET)
@@ -11,11 +13,14 @@ all: $(TARGET)
 clean:
 	rm -f $(TARGET) $(OBJS)
 
-$(TARGET): $(OBJS)
-	ar r test_tmp.a $(OBJS)
-	ar crsT $(TARGET) ../../yges_c.a test_tmp.a
+$(TARGET): $(INTERMED) $(LINKAGES)
+	rm -f $(TARGET)
+	ar crsT $(TARGET) $(LINKAGES) $(INTERMED)
 
-%.o: %.c
+$(INTERMED): $(OBJS) $(LINKAGES)
+	ar r $(INTERMED) $(OBJS)
+
+%.o: %.c $(LINKAGES)
 	$(CC) -o $@ -c $< $(CFLAGS)
 
-.PHONY: clean
+.PHONY: all clean

